FlowCrypt/Functionality/Services/EnterpriseServerApi.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  EnterpriseServerApi.swift
+//  FlowCrypt
+//
+//  Created by Yevhen Kyivskyi on 05.06.2021.
+//  Copyright © 2021 FlowCrypt Limited. All rights reserved.
+//
+
+import Promises
+
+protocol EnterpriseServerApiType {
+    func getActiveFesUrl(for email: String) -> Promise<String?>
+}
+
+class EnterpriseServerApi: EnterpriseServerApiType {
+
+    private enum Constants {
+        static let getActiveFesTimeout: TimeInterval = 4
+
+        static let serviceKey = "service"
+        static let serviceNeededValue = "enterprise-server"
+    }
+
+    func getActiveFesUrl(for email: String) -> Promise<String?> {
+        Promise<String?> { resolve, _ in
+            guard let userDomain = email.recipientDomain,
+                  !Configuration.excludedDomains.contains(userDomain) else {
+                resolve(nil)
+                return
+            }
+            let urlString = "https://fes.\(userDomain)/"
+            let request = URLRequest.urlRequest(
+                with: "\(urlString)api/",
+                method: .get,
+                body: nil
+            )
+
+            let response = try awaitPromise(URLSession.shared.call(request, tolerateStatus: [404]))
```

Comment:
It should tolerate basically all errors and return nil in those situations (since we don't know if there's a server running there or not until we try, for most users, no server will be running there, so we want to return nil graciously)

Reply:
done